meson.build

```diff
@@ -0,0 +1,92 @@
+project('lcm', 'c', 'cpp',
+  version : '1.5.0',
+  default_options : ['buildtype=release',
+                     'c_std=gnu99',
+                     'warning_level=0',
+                     'c_args=-Wno-format-zero-length'])
+
+# Dependencies
+glib_dep = dependency('glib-2.0')
+
+# set windows dependencies
+unix = host_machine.system() != 'windows'
+windows = not unix
+if windows
+  cpp = meson.get_compiler('cpp')
+  wsock32 = cpp.find_library('wsock32')
+  ws2_32 = cpp.find_library('ws2_32')
+endif
+
+### Core modules ###
+# lcm-gen compiles with warnings that cannot be disabled when lcm is used
+# as a Meson subproject in a parent project with more warnings enabled. Allow
+# parent projects to disable compilation of lcm-gen to fix warnings as needed.
+subdir('lcm')
+subdir('lcm-logger')
+if not get_option('lcm_enable_lcmgen').disabled()
+  subdir('lcmgen')
+endif
+
+# Not implemented yet
+# Documentation (Main, C/C++, .NET)
+# subdir('docs')
+
+# Doesn't seem to compile examples but compiles tests?
+# We don't want to compile examples, which makes this option misleading, but it
+# is how it is in the CMakeLists.txt file.
+# Examples
+if not get_option('lcm_enable_examples').disabled()
+  subdir('liblcm-test')
+endif
+
+### Extra modules (Work in progress) ###
+# Python
+# Currently doesn't compile for Windows
+subdir('lcm-python')
```

Comment:
I agree that it would be ideal for all of them to be checked in this meson.build instead of the subdirectory, but this inconsistency was done to include extra logic for detailed messages, because those aren't currently supported, without cluttering the main meson.build. I can work on adding extra documentation for the rationale. Where should it go?

Reply:
Ah, that makes sense.  If that's the reason for all of them, then I'd just add a comment in the top-level `meson.build` to just give a reader that heads-up.

Reply:
Potentially relevant: one small advantage of doing it in the main `meson.build` rather than in the subdirectory is it results in less indentation and potentially more readable logic. In the former, you're just wrapping the `subdir` call in an `if` statement. In the latter, you're wrapping most of your build code in an `if` statement. I think either way is fine though, personally.

Reply:
Added rationale in top level meson.build f2e0c69da4e805fe2d0fbfcba42168cdb2c780dc